app/models/ckey2bibframe.rb

```diff
@@ -24,19 +24,22 @@ def marc21_to_xml
       Rails.logger.warn('Unable to connect to Symphony via SSH.')
       Rails.logger.error(e)
     end
-    File.write(marcxml_file, marcxml)
+    marcxml_file
+    @marcxml_file.write(marcxml)
     Nokogiri::XML(marcxml, &:noblanks)
   end
 
   def marcxml_to_bibframe
+    @marcxml_file.read
     bf2_xsl = 'loc_marc2bibframe2/xsl/marc2bibframe2.xsl'
-    command = "xsltproc --stringparam baseuri #{baseuri} #{bf2_xsl} #{marcxml_file}"
+    command = "xsltproc --stringparam baseuri #{baseuri} #{bf2_xsl} #{@marcxml_file.path}"
     bibframe = execute_command(command).force_encoding('UTF-8')
-    FileUtils.rm(marcxml_file)
+    @marcxml_file.close
+    @marcxml_file.delete
     Nokogiri::XML(bibframe, &:noblanks)
```

Comment:
All the references to `@marcxml_file` can be replaced with the method call to `marcxml_file` without the `@`.  The method will take of everything and return the instance variable every time.  All the calls on that object can be chained just as if it is an instance var-obj.